numpy/core/tests/test_multiarray.py

```diff
@@ -3554,6 +3554,13 @@ def test_field_rename(self):
         dt.names = ['p', 'q']
         assert_equal(dt.names, ['p', 'q'])
 
+    def test_multiple_field_name_occurance(self):
+        def test_assign():
+            dtype = np.dtype([("A", "f8"), ("B", "f8"), ("A", "f8")])
+
+        # Error raised when multiple fields have the same name
+        assert_raises(ValueError, test_assign)
+
     if sys.version_info[0] >= 3:
         def test_bytes_fields(self):
```

Comment:
Could you add the unicode test you showed in the comments here, for Python 3 versions only?